DOC-8272-C4-Apple --Swift/ObjC N1QL Code Snippets

ibsoln opened the pull request:
https://issues.couchbase.com/browse/DOC-8272

DOC-9047-C4-Code

jayahariv commented on modules/objc/examples/code_snippets/SampleCodeTest.m:
```diff
@@ -1804,7 +1804,48 @@ - (void) dontTestQuerySyntaxJson {
+    */
 
+    // tag::query-syntax-n1ql[]
+    NSError *error;
+
+    CBLDatabase *db == argDB;
```
do you think this two variable declaration are important, and need to be displayed?

I mean, we could keep this above the `tag::query-syntax-n1ql[]`

ibsoln replied:
DONe

jayahariv commented on modules/objc/examples/code_snippets/SampleCodeTest.m:
```diff
@@ -1804,7 +1804,48 @@ - (void) dontTestQuerySyntaxJson {
+
+    CBLDatabase *db == argDB;
+
+    NSString *n1qlstr = "SELECT * FROM _ WHERE type = \"hotel\""; // <.>
```
not sure, should we promote users to write N1QL query using `_` or with a valid database name?

ibsoln replied:
As far as I am aware Jay _ is fine to use; may change when collections and scopes are supported

jayahariv commented on modules/objc/examples/code_snippets/SampleCodeTest.m:
```diff
@@ -1804,7 +1804,48 @@ - (void) dontTestQuerySyntaxJson {
+    */
+
+    // tag::query-syntax-n1ql-params[]
+    NSError *error;
+
+    CBLDatabase *db = argDB;
```
same, like above.

ibsoln replied:
Done

jayahariv commented on modules/objc/examples/code_snippets/SampleCodeTest.m:
```diff
@@ -1804,7 +1804,48 @@ - (void) dontTestQuerySyntaxJson {
+
+    CBLDatabase *db = argDB;
+
+    NSString *n1qlstr = "SELECT * FROM _ WHERE type = $type"; // <.>
+
+    CBLQuery* thisQuery = [db createQuery: n1qlstr];
+
+    CBLQueryParameters* n1qlparams = [[CBLQueryParameters alloc] init];
+    [params setString: @"hotel" forName: @"type"]; // <.>
+
+    thisQuery.parameters = n1qlparams;
```
don't think this is correct! 


But something like the below is a way to set a dynamic variable value to a N1QL query. 
```
NSString* typeVar = @"hotel";
NSString *n1qlstr = [NSString stringWithFormat: @"SELECT * FROM dbName WHERE type = %@", typeVar];
```

ibsoln replied:
Hm. Neat trick. Not sure Jay. I thought that we were supposed to use the Query parameters ... maybe both work, but which do we want people using? I'll follow this up and change asap.

jayahariv replied:
@ibsoln  I don't think this code will work, using QueryParams!

jayahariv reviewed (changes requested):
please check the comments.

ibsoln committed:
DOC-8272-C4-C --N1QL Code Snippets
https://issues.couchbase.com/browse/DOC-8272

DOC-8272-C4-Apple --Swift/ObjC N1QL Code Snippets
https://issues.couchbase.com/browse/DOC-8272

DOC-9047-C4-Code

ibsoln committed:
DOC-8272-C4Apple --Swift/ObjC N1QL Feedback https://issues.couchbase.com/browse/DOC-8272
